linux/src/threadpool_linux.c

```diff
@@ -716,6 +757,8 @@ void threadpool_timer_destroy(TIMER_INSTANCE_HANDLE timer)
         {
             // Do Nothing
         }
+        /* Codes_SRS_THREADPOOL_LINUX_45_012: [ threadpool_timer_cancel shall call ThreadAPI_Sleep to allow timer resources to clean up. ]*/
```

Comment:
>Codes_SRS_THREADPOOL_LINUX_45_012 [](http://example.com/codeflow?start=11&length=33)

I guess we should have a comment here explaining why this is needed (the timer thread can be scheduled even though we called timer_delete.

Also I had no idea the win32 API's were so much better for this, `WaitForThreadpoolTimerCallbacks` makes this much easier in Win32!

Reply:
Added a comment before the wait which explains why it's there.